src/client/pages/ApplyPage/components/ResponseSection.tsx

```diff
@@ -164,98 +172,113 @@ class ResponseSection extends ApplyPageSection<ResponseSectionFormData, Response
 
     return (
       <form onSubmit={handleSubmit}>
-        {options.foodOption && FormFields.createRow(
-          FormFields.createColumn('col-sm-12',
-            FormFields.createLabel('What kind of food would you like to see ' +
-              'at the hackathon?', false),
-            FormFields.createTextArea('food', 'Healthy Snacks and Drinks')
-          )
-        )}
-
-        {FormFields.createRow(
-          FormFields.createColumn('col-sm-12',
-            FormFields.createLabel('Dietary Restrictions', false),
-            FormFields.createTextArea('diet', 'Dietary Restrictions')
-          )
-        )}
+        {options.foodOption &&
+          <ApplicationRow>
+            <ApplicationCol className="col-sm-12">
+              <ApplicationLabel required={false}>
+                What kind of food would you like to see at the event?
+              </ApplicationLabel>
+              <ApplicationTextArea name="food" placeholder="Healthy Snacks and Drinks" />
+            </ApplicationCol>
+          </ApplicationRow>}
+        <ApplicationRow>
+          <ApplicationCol className="col-sm-12">
+            <ApplicationLabel required={false}>Dietary Restrictions</ApplicationLabel>
+            <ApplicationTextArea name="diet" placeholder="Dietary Restrictions" />
+          </ApplicationCol>
+        </ApplicationRow>
 
         {options.requireWhyThisEvent &&
-          FormFields.createColumn('col-sm-12',
-            FormFields.createLabel(`Why Do You Want To Attend ${event.name}?`, true),
-            FormFields.createTextArea('whyEventResponse', 'Your Response...')
-          )
+          <ApplicationCol className="col-sm-12">
+            <ApplicationLabel required={true}>
+              <>Why Do You Want To Attend {event.name}?</>
```

Comment:
Does this require the empty blocks? I guess because it needs to be a React component to pass the type checking?

Reply:
Yes! We are passing in `Why Do you want to attend {event.name}?` to `ApplicationLabel`, and the use of `{event.name}` requires the surrounding text to be a React node - so I wrapped it in a Fragment, which is the intended use case anyway.